openvdb/openvdb/tree/InternalNode.h

```diff
@@ -2886,173 +2856,6 @@ InternalNode<ChildT, Log2Dim>::combine2(const InternalNode& other, const OtherVa
-{
-    // Allow the two nodes to have different ValueTypes, but not different dimensions.
-    static_assert(OtherNodeT::NUM_VALUES == NodeT::NUM_VALUES,
-        "visit2() requires nodes to have the same dimensions");
-    static_assert(OtherNodeT::LEVEL == NodeT::LEVEL,
-        "visit2() requires nodes to be at the same tree level");
-
-    typename NodeT::ValueType val;
-    typename OtherNodeT::ValueType otherVal;
-
-    ChildAllIterT iter = self.beginChildAll();
-    OtherChildAllIterT otherIter = other.beginChildAll();
-
-    for ( ; iter && otherIter; ++iter, ++otherIter)
-    {
-        const size_t skipBranch = static_cast<size_t>(op(iter, otherIter));
-
-        typename ChildAllIterT::ChildNodeType* child =
-            (skipBranch & 1U) ? nullptr : iter.probeChild(val);
-        typename OtherChildAllIterT::ChildNodeType* otherChild =
-            (skipBranch & 2U) ? nullptr : otherIter.probeChild(otherVal);
-
-        if (child != nullptr && otherChild != nullptr) {
-            child->visit2Node(*otherChild, op);
-        } else if (child != nullptr) {
-            child->visit2(otherIter, op);
-        } else if (otherChild != nullptr) {
-            otherChild->visit2(iter, op, /*otherIsLHS=*/true);
-        }
-    }
-}
-
-
-////////////////////////////////////////
-
-
-template<typename ChildT, Index Log2Dim>
-template<typename OtherChildAllIterType, typename VisitorOp>
-inline void
-InternalNode<ChildT, Log2Dim>::visit2(OtherChildAllIterType& otherIter,
-    VisitorOp& op, bool otherIsLHS)
-{
-    doVisit2<InternalNode, VisitorOp, ChildAllIter, OtherChildAllIterType>(
-        *this, otherIter, op, otherIsLHS);
-}
-
-
-template<typename ChildT, Index Log2Dim>
-template<typename OtherChildAllIterType, typename VisitorOp>
-inline void
-InternalNode<ChildT, Log2Dim>::visit2(OtherChildAllIterType& otherIter,
-    VisitorOp& op, bool otherIsLHS) const
-{
-    doVisit2<const InternalNode, VisitorOp, ChildAllCIter, OtherChildAllIterType>(
-        *this, otherIter, op, otherIsLHS);
-}
-
-
-template<typename ChildT, Index Log2Dim>
-template<typename NodeT, typename VisitorOp, typename ChildAllIterT, typename OtherChildAllIterT>
-inline void
-InternalNode<ChildT, Log2Dim>::doVisit2(NodeT& self, OtherChildAllIterT& otherIter,
-    VisitorOp& op, bool otherIsLHS)
-{
-    if (!otherIter) return;
-
-    const size_t skipBitMask = (otherIsLHS ? 2U : 1U);
-
-    typename NodeT::ValueType val;
-    for (ChildAllIterT iter = self.beginChildAll(); iter; ++iter) {
-        const size_t skipBranch = static_cast<size_t>(
-            otherIsLHS ? op(otherIter, iter) : op(iter, otherIter));
-
-        typename ChildAllIterT::ChildNodeType* child =
-            (skipBranch & skipBitMask) ? nullptr : iter.probeChild(val);
-
-        if (child != nullptr) child->visit2(otherIter, op, otherIsLHS);
-    }
-}
-
-
-////////////////////////////////////////
-
-
 template<typename ChildT, Index Log2Dim>
```

Comment:
They have been replaced by the DynamicNodeManager. This class offers all the same functionality and benefits as the visitor classes, but is significantly more performant and doesn't pollute the Tree hierarchy with lots of additional class members that slows down compile times. We have worked through all the edge cases and concerns that have been raised since we introduced this class more than two years ago (const trees, grain sizes, sequential visitor patterns -addressed in adding `tools::visitNodesDepthFirst`) and lots of our existing tools have long since been rewritten or adapted to make use of the DynamicNodeManager (this includes activate, merge, count, evalMinMax, CSG, fast sweeping, etc) so the visitor classes have been deprecated and unused since at least v9.0. The decision was taken a while ago to drop these visitor methods on the next major API bump which is v10.0.